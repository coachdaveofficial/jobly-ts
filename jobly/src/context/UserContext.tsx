import React from "react";
import {User} from '../types/types'

/** Context: provides currentUser object and setter for it throughout app. */

how to type?

const UserContext = React.createContext<any>(null);



export default UserContext;
