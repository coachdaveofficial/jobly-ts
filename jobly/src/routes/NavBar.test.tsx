import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBarFunc from "./NavBar";
import UserContext from "../context/UserContext";

it("matches the snapshot", () => {
    const {asFragment} = render(
        <MemoryRouter>
            <NavBarFunc logout={() => {}} />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", () => {
    const {getByText} = render(
        <MemoryRouter>
            <NavBarFunc logout={() => {}} />
        </MemoryRouter>
    );

    expect(getByText("Jobly")).toBeInTheDocument();
    expect(getByText("Login")).toBeInTheDocument();
    expect(getByText("Signup")).toBeInTheDocument();
});

it("renders navbar with logged in links when user is logged in", () => {
    const currentUser = {username: "testuser", firstName: "", lastName: "", password: "", email: "", isAdmin: false}
    const setCurrentUser = () => {}
    const { getByText } = render(
      <UserContext.Provider value={{currentUser, setCurrentUser} }>
        <MemoryRouter>
        <NavBarFunc logout={() => {}} />
        </MemoryRouter>
        
      </UserContext.Provider>
    );
  
    expect(getByText("Companies")).toBeInTheDocument();
    expect(getByText("Jobs")).toBeInTheDocument();
    expect(getByText("Profile")).toBeInTheDocument();
    expect(getByText("Logout testuser")).toBeInTheDocument();
    expect(getByText("Jobly")).toBeInTheDocument();
  });