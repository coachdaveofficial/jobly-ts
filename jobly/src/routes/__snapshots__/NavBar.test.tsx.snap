// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: lightblue;"
  >
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/login"
            >
              Login
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/signup"
            >
              Signup
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
